Source/BluetoothManager.swift

```diff
@@ -342,22 +342,29 @@ public class BluetoothManager {
     /// - Parameter peripheral: `Peripheral` which is monitored for connection.
     /// - Returns: Observable which emits next events when `peripheral` was connected.
     public func monitorConnection(for peripheral: Peripheral) -> Observable<Peripheral> {
-        return monitorPeripheral(on: delegateWrapper.rx_didConnectPeripheral, peripheral: peripheral)
+        return monitorPeripheral(
+            on: delegateWrapper.rx_didConnectPeripheral.map { ($0, nil) },
+            peripheral: peripheral
+        ).map { $0.0 }
     }
 
     /// Emits `Peripheral` instance when it's disconnected.
     /// - Parameter peripheral: `Peripheral` which is monitored for disconnection.
-    /// - Returns: Observable which emits next events when `peripheral` was disconnected.
-    public func monitorDisconnection(for peripheral: Peripheral) -> Observable<Peripheral> {
-        return monitorPeripheral(on: delegateWrapper.rx_didDisconnectPeripheral.map { $0.0 }, peripheral: peripheral)
+    /// - Returns: Observable which emits next events when `peripheral` was disconnected or error event when occured.
+    public func monitorDisconnection(for peripheral: Peripheral) -> Observable<(Peripheral, BluetoothError?)> {
+        return monitorPeripheral(
+            on: delegateWrapper.rx_didDisconnectPeripheral.map { (_, error) in
+                throw BluetoothError.peripheralDisconnectionFailed(peripheral, error)
+            }, peripheral: peripheral
+        )
     }
 
-    func monitorPeripheral(on peripheralAction: Observable<CBPeripheral>, peripheral: Peripheral)
-        -> Observable<Peripheral> {
+    func monitorPeripheral(on peripheralAction: Observable<(CBPeripheral, BluetoothError?)>, peripheral: Peripheral)
```

Comment:
I'm not sure if this function is worth to share it between connected/disconnected events.

Reply:
in my opinion it is. If we would like to separate it instead of two easy maps we would have to implement logic of filtering and mapping on stream  of `pheripheralAction` with ensuring poweredOn state in both function `monitorConnection` and `monitorDisconnection`